### PR TITLE
Add python3 and django 2.2 support

### PR DESCRIPTION
**Pull request by jlward**

### CODE REVIEW

**Comment by jlward** [comment]
Travis is doing something wonky. But tests are passing with tox (locally). There is a ton of flake8 in there. And there is a lot of `six` related changes.

File: tox.ini [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
+    py27-django111,
+    py37-django{111,220}
 
 [testenv]
 commands = {envpython} run_tests.py
 deps =
     nose
     django_nose
+    six
     django18: Django>=1.8, <1.9
```
Comment (bocribbz) [inline tox.ini]: `django18` can probably be removed too.

**Comment by bocribbz** [comment]
why is travis trying to run tests for `py27-django220`?
something is not right....

File: .travis.yml [hunk 2]
```diff
@@ -2,8 +2,9 @@ language: python
 python:
   - "2.7"
 env:
-  - TOX_ENV=py27-django18
   - TOX_ENV=py27-django111
+  - TOX_ENV=py27-django220
```
Comment (bocribbz) [inline .travis.yml]: ah this is why it's running `py27-django220` this should probablt be `py37-django111` instead?

File: .travis.yml [hunk 3]
```diff
@@ -2,8 +2,9 @@ language: python
 python:
   - "2.7"
```
Comment (bocribbz) [inline .travis.yml]: we need python `3.7` defined in here too

**Comment by jlward** [comment]
Travis cannot use python3.7. For whatever reason.

**Comment by bocribbz** [comment]
https://github.com/travis-ci/travis-ci/issues/9815

**Comment by bocribbz** [comment]
also, this seems to be the official "workaround" https://github.com/travis-ci/travis-ci/issues/9069#issuecomment-425720905

File: .travis.yml [hunk 4]
```diff
@@ -1,9 +1,11 @@
 language: python
+dist: xenial
 python:
-  - "2.7"
+  - "3.7"
```
Comment (bocribbz) [inline .travis.yml:4]: i think you need both if you want to run some tests on 2.7 and some tests on 3.7

Comment (jlward) [reply .travis.yml:4]: Negative. Having both just ran all the tests twice. And the python3 tests via python2 don't work at all.

Comment (jlward) [reply .travis.yml:4]: I tried that, it didn't work.

**Review by bocribbz — APPROVED** [review]

**Pull request merged**